### ISSUE TITLE
Enhance code quality and maintainability by using Sonarcloud

### ISSUE THREAD

**Issue by michael-schnell** [quarkusio/quarkus#8587]
**Description**
Currently there is a lack of checks for code quality and style. This could be enhanced by some static code analysis.

**Implementation ideas**
Quarkus should add a Maven plugin to send analysis data to [SonarCloud.io](https://sonarcloud.io/) which is free for open source projects. This will help all developers to enhance code quality and improve stability and maintainability. Plugins for IDEs (like Eclipse) are also available.

Other projects like SmallRye already use this kind of setup:
https://sonarcloud.io/dashboard?id=smallrye_smallrye-open-api

**Comment by nimo23** [comment]
Another alternative is using https://lgtm.com/ as suggested in https://github.com/quarkusio/quarkus/issues/5892.

**Comment by michael-schnell** [comment]
Interesting. Never heard of that platform before. All corporate projects I have been involved so far use SonarQube or SonarCloud. 

I suggested SonarCloud also, because we could simply copy the Maven plugin setup from SmallRye. Only thing is someone to create an account for Quarkus and add the credentials to the build platform. Then everything should work almost out of the box.

**Comment by nimo23** [comment]
I use both because they complete each other and are not a substitution. LGTM is also completely free for open source projects. LGTM has strengths compared to SonarQube. LGTM can analyze the code depper. Compare yourself https://lgtm.com/projects/g/wildfly/wildfly?mode=list.

**Comment by michael-schnell** [comment]
Yeah, but I would be glad if at least one static code analysis is in place ;-) 

@emmanuelbernard Who can create an account for Quarkus? I could create a PR based on the setup from SmallRye.

**Comment by nimo23** [comment]
Yes, would be good to have SonarCloud. This would also save time for code reviewers..
